src/WaveClip.cpp

```diff
@@ -1789,3 +1717,233 @@ const wxString& WaveClip::GetName() const
 {
    return mName;
 }
+
+void WaveClip::ClearSequence(double t0, double t1)
+{
+    auto clip_t0 = std::max(t0, GetSequenceStartTime());
+    auto clip_t1 = std::min(t1, GetSequenceEndTime());
+
+    auto s0 = TimeToSequenceSamples(clip_t0);
+    auto s1 = TimeToSequenceSamples(clip_t1);
+
+    if (s0 != s1)
+    {
+        // use Strong-guarantee
+        GetSequence()->Delete(s0, s1 - s0);
+
+        // use No-fail-guarantee in the remaining
+
+        // msmeyer
+        //
+        // Delete all cutlines that are within the given area, if any.
+        //
+        // Note that when cutlines are active, two functions are used:
+        // Clear() and ClearAndAddCutLine(). ClearAndAddCutLine() is called
+        // whenever the user directly calls a command that removes some audio, e.g.
+        // "Cut" or "Clear" from the menu. This command takes care about recursive
+        // preserving of cutlines within clips. Clear() is called when internal
+        // operations want to remove audio. In the latter case, it is the right
+        // thing to just remove all cutlines within the area.
+        //
+
+        // May DELETE as we iterate, so don't use range-for
+        for (auto it = mCutLines.begin(); it != mCutLines.end();)
+        {
+            WaveClip* clip = it->get();
+            double cutlinePosition = GetSequenceStartTime() + clip->GetSequenceStartTime();
+            if (cutlinePosition >= t0 && cutlinePosition <= t1)
+            {
+                // This cutline is within the area, DELETE it
+                it = mCutLines.erase(it);
+            }
+            else
+            {
+                if (cutlinePosition >= t1)
+                {
+                    clip->Offset(clip_t0 - clip_t1);
+                }
+                ++it;
+            }
+        }
+
+        // Collapse envelope
+        auto sampleTime = 1.0 / GetRate();
+        GetEnvelope()->CollapseRegion(t0, t1, sampleTime);
+    }
+    
+
+    MarkChanged();
+}
+
+sampleCount WaveClip::TimeToSamples(double time) const noexcept
+{
+    return sampleCount(floor(time * mRate + 0.5));
+}
+
+double WaveClip::SamplesToTime(sampleCount s) const noexcept
+{
+    return s.as_double() / mRate;
+}
+
+void WaveClip::SetSilence(sampleCount offset, sampleCount length)
+{
+    GetSequence()->SetSilence(sampleCount(floor(GetTrimLeft() * GetRate())) + offset, length);
+    MarkChanged();
+}
+
+sampleCount WaveClip::GetSequenceSamplesCount() const
+{
+    return mSequence->GetNumSamples();
+}
+
+double WaveClip::GetPlayStartTime() const noexcept
+{
+    return mSequenceOffset + mTrimLeft;
+}
+
+void WaveClip::SetPlayStartTime(double time)
+{
+    mSequenceOffset = time - mTrimLeft;
```

Comment:
If this function instead called through to WaveClip::SetSequenceStartTime(), which does change the envelope offset, then the bug last mentioned would be fixed.

Reply:
Thanks!